Sources/AppStoreServerLibrary/ChainVerifier.swift

```diff
@@ -150,31 +149,22 @@ final class AppStoreOIDPolicy: VerifierPolicy {
 
 final class Requester: OCSPRequester {
     func query(request: [UInt8], uri: String) async -> X509.OCSPRequesterQueryResult {
-        let url = URL(string: uri)!
-        var urlRequest = URLRequest(url: url)
-        urlRequest.httpMethod = "POST"
-        urlRequest.setValue("application/ocsp-request", forHTTPHeaderField: "Content-Type")
-        urlRequest.httpBody = Foundation.Data(request)
-        
         do {
-            let data: Foundation.Data = try await withCheckedThrowingContinuation() { c in
-                let urlSessionDataTask = URLSession.shared.dataTask(with: urlRequest) {data, response, error in
-                    if let e = error {
-                        c.resume(throwing: e)
-                        return
-                    }
-                    guard let unwrappedData = data else {
-                        c.resume(throwing: OCSPFetchError())
-                        return
-                    }
-                    c.resume(returning: unwrappedData)
-                }
-                urlSessionDataTask.resume()
+            let httpClient = HTTPClient()
+            defer {
+                try? httpClient.syncShutdown()
+            }
+            var urlRequest = HTTPClientRequest(url: uri)
+            urlRequest.method = .POST
+            urlRequest.headers.add(name: "Content-Type", value: "application/ocsp-request")
+            urlRequest.body = .bytes(request)
+            let response = try await httpClient.execute(urlRequest, timeout: .seconds(30))
+            var body = try await response.body.collect(upTo: 1024 * 1024)
+            guard let data = body.readData(length: body.readableBytes) else {
+                return .nonTerminalError(OCSPFetchError())
```

Comment:
Why is this not a terminalError? If we are unable to read data from the OCSP request, wouldn't that be an unrecoverable error?

Reply:
Err I seem to remember it being caught elsewhere but yeah you're right. I'll update